examples/baroclinic_channel/baroclinic_channel.py

```diff
@@ -0,0 +1,190 @@
+"""
+Baroclinic channel test case following [1]
+
+The domain extends from latitude 30 N to 45 N and is 20 deg long in zonal
+direction. Periodic boundaries are used in eastern and western boundaries.
+Depth is constant 1600 m. Mesh resolution is 1/6 deg in zonal and 1/7 deg
+in meridional directions. 24 evenly-spaced vertical levels are used. In [2]
+resolutions 1/3, 1/6, 1/9, 1/12, 1/24, and 1/36 deg are used.
+
+Here the domain is approximated with a 1600-by-1600 km box in Cartesian
+coordinates with the following resolutions
+
+n_elem   delta x (m)    deg  levels
+-----------------------------------
+ 43      37209.30      ~1/3      20
+ 86      18604.65      ~1/6      20
+ 172     9302.326     ~1/12      20
+ 344     4651.163     ~1/24      20
+
+Initial temperature is defined by gradients dT/dy=-5e-6 degC/m and
+dT/dz=8.2e-3 degC/m. Maximum surface temperature is 25 degC.
+Salinity is constant 35 psu.
+
+Vertical viscosity and diffusivity are set to 1e-3 and 1e-5 m2/s, respectively.
+Pacanowski-Philander vertical mixing scheme is used for temperature with max.
+diffusivity 0.01 m2/s. Horizontal diffusivity 30 m2/s is used. Horizontal
+viscosity depends on the advection scheme (typ. 100 m2/s). Bottom drag
+coefficient is Cd=0.0025.
+
+[1] Danilov (2012). Two finite-volume unstructured mesh models for
+large-scale ocean modeling. Ocean Modelling, 47:14-25.
+[2] Danilov and Wang (2015). Resolving eddies by local mesh refinement.
+ Ocean Modelling, 93:75-83.
+"""
+
+from thetis import *
+comm = COMM_WORLD
+
+physical_constants['rho0'] = 1020.
+
+refinement = 1  # normal = 4
+lx = ly = 1600e3
+nx = ny = 43*refinement
+delta_x = lx/nx
+delta_y = ly/ny
+mesh2d = PeriodicRectangleMesh(nx, ny, lx, ly, direction='x')
+depth = 1600.
+nlayers = 10  # FIXME
```

Comment:
Does this need fixing indeed?

Reply:
Yes, Danilov uses 24 levels, I used 10 for testing. I'll revise this upstream.